o2a/converter/workflow.py

```diff
@@ -55,6 +57,15 @@ def __init__(
             "from airflow.utils.trigger_rule import TriggerRule",
             "from airflow.utils import dates",
         }
+        self.library_folder = os.path.join(self.input_directory_path, HDFS_FOLDER, LIB_FOLDER)
+        self.jar_files = self.get_lib_files(extension=".jar")
```

Comment:
I have a doubt whether this is the best place to execute this code. I think we should limit the operations performed in the constructor.

Reply:
Any proposal?

Reply:
I think it is definitely needed here because of #301 and #302 . Support for shared lib .so and .jar files is common across all actions but we have not implemented it so far. Therefore it should be in workflow in my opinion.

Reply:
I have no doubt if this field should exist here. I think it is worth to set this field in the convert method. I think that we should limit the operations performed in the constructor only to setting values.

I think it can be in a separate file. This is not called by any other class, so placing this method in the main object of the entire application may can lead to misunderstanding.

Reply:
This class has only current roles - storing information and I would like it to continue. Other methods include more advanced setters and getters, or system methods.

Reply:
Let's talk about it in-person. I do not understand what you propose.

Reply:
@mik-laj  - I have a proposal.

In order to makeyour proposal (and our code) more understandable, maybe you should update the docstrings of each of the classes so that we are clear about what each class's purpose means. You seem to comment having some purpose in mind, but I clearly lack the insight of what the Workflow class true purpose is (but you seem to know it). Currently docstring for Workflow is:

"""Class for Workflow"""

IF (I do not know) I understand correctly you still remember that Workflow was a container for some small amount of data (but I am guessing now). This used to be true but now it contains a lot more methods that really encapsulate quite a number of methods. I actually fixed an old incorrect information in Workflow - see  9ebe183 where pylint was disabled and comment that Workflow has no public methods (this used to be True but it is not any more). But I am guessing because docstrings and comments are not updated to reflect the current Workflow purpose.

As I understand it - I think Workflow is pretty decent place to store information about related workflow lib jar files. 

Also I think you have quite clear understanding of the architecture - maybe you should describe the scope and "contract" of each of the related classes (such as Workflow)- what it does, what the purpose of this class is ideally in the docstring. You seem to know best what the responsibility, scope and "contract" of each of the classes are, so you are the best person to do so.

Maybe you can make a PR that describes clearly in docstrings what each class is about and then we can discuss how lib_jars fit it ? That would be great way to start discussion.